### DISCUSSION TITLE
Self signed certificate

### DISCUSSION THREAD

**Discussion by michael-gvdw** [deployKF/deployKF#195] [Questions & Answers]
Is it possible to pass a self signed certificate when trying to connect to en external object store in the configuration of deploykf?

**Comment by thesuperzapper** [comment]
@michael-gvdw did you run into a specific problem, or is this a general question?

If you have a self-signed object store already set up, please test it to see if anything fails so we can document it.

Just follow the normal docs for [connecting an object store](https://www.deploykf.org/guides/external/object-store/#connect-an-external-object-store) and tell us which parts don't work (if any).

**Comment by michael-gvdw** [comment]
@thesuperzapper when i alter the values in the apps of apps yaml file in which i provide the neccessary configuration in order to connect to an external object store at some point during the sync script i get the following error:
```
2024-08-19T08:35:56+02:00  kyverno.io                    ClusterPolicy                              patch-kfp-v2-pods-d81b8cd1-team-1-prod                      Synced                            
2024-08-19T08:35:57+02:00   apps  Deployment      team-1  ml-pipeline-ui-artifact    Synced  Healthy              deployment.apps/ml-pipeline-ui-artifact configured
2024-08-19T08:35:59+02:00   apps  Deployment  team-1-prod  ml-pipeline-ui-artifact    Synced  Healthy              deployment.apps/ml-pipeline-ui-artifact configured
2024-08-19T08:36:14+02:00   apps  Deployment    kubeflow        ml-pipeline-ui    Synced  Healthy              deployment.apps/ml-pipeline-ui configured
FATA[0300] rpc error: code = Canceled desc = context canceled
```

I also have attached the apps of apps yaml file.
[deploykf-app-of-apps.txt](https://github.com/user-attachments/files/16656814/deploykf-app-of-apps.txt)

**Reply by michael-gvdw** [reply]
@thesuperzapper  With some further investiongation i found the following. The pod ml-pipeline-xxxx-xxxx in ns kubeflow has the status CrashLoopBackOff. When viewing the logs (as assumed) I see the following error "Failed to check if Minio bucket exists. Error: Get "https://pllbltst.catharinazkh.local/kubeflow-test/?location=": tls: failed to verify certificate: x509: certificate signed by unknown authority"

How could i either trust the certificate or just not verify it and trust it blindly?

**Comment by michael-gvdw** [comment]
I managed to get it working (kinda). Since it was a ssl error I created a configmap with the .crt file. After I edited the Deployment of "ml-pipeline" in which i mounted the configmap as a volume. This way the ssl crt was available to the pod and it was able to connect successfully to the external object store.

I have two follow up questions:

1. Which resource would I need to alter in the deploykf repo in order to keep using the sync_script? (now i first deploy and edit the deployed resource)

2. For some reason unkown to me the pipeline is registerd as failed in the UI but the kubernetes pod has status completed and also the s3 bucket is populated with data of the pipeline, is there a known issue about this behaviour?
